### ISSUE TITLE
Expressions for first arg of clojure.test/testing generates missing test assertion warnings

### ISSUE THREAD

**Issue by khardenstine** [clj-kondo/clj-kondo#2083]
**version**

2023.05.18

**problem**

Using an expression as the first argument of clojure.test/testing generates "missing test assertion" warnings

**repro**

```clj
(ns repro
  (:require
    [clojure.test :refer [deftest testing is]]))

(deftest foo
  (testing (str "str " "some " "strings")
    (is (= 1 2))))
```

```bash
clj-kondo --lint repro.clj
repro.clj:6:12: warning: missing test assertion
linting took 9ms, errors: 0, warnings: 1
```

**expected behavior**

no warning

**Comment by borkdude** [comment]
Thanks for this repro. You can work around this with:

``` clojure
(ns repro
  (:require
   [clojure.test :refer [deftest testing is]]))

(deftest foo
  (testing #_:clj-kondo/ignore (str "str " "some " "strings")
    (is (= 1 2))))
```

for the time being.

**Comment by borkdude** [comment]
Explanation how this was introduced:

``` clojure
(testing (not (= 1 2)))
```

did not trigger a warning, so all "pure" functions like `not` that are not wrapped in an `is` now report a `:missing-test-assertion` (whereas `(testing "foo" (true? (= 1 2)))` did trigger such a warning, but the first argument to `testing` should not be considered.
